### ISSUE TITLE
We should consider creating a .jar distribution

### ISSUE THREAD

**Issue by devos50** [Tribler/kotlin-ipv8#53]
When one currently wants to use this repository, there is no other option than to add this project as submodule, as done by the [superapp](https://github.com/Tribler/trustchain-superapp). However, Gradle does not really support this kind of architecture and support for it is whacky. The "proper" way to re-use the code in this repository is by building a `.jar` file and including it in the project (the same with depending on Python modules). Therefore, it might be a good idea to build this file when merging a PR.

**Comment by KoningR** [comment]
Agreed! Using this repository is currently much more difficult and time-consuming than necessary. It also doesn’t help that the supported Java and Gradle versions are rather old.

**Comment by synctext** [comment]
solid! just make sure we keep supporting old and even ancient Android phones.
